src/Libraries/RevitNodes/Elements/ModelCurve.cs

```diff
@@ -79,11 +79,13 @@ private void InitModelCurve(Autodesk.Revit.DB.Curve crv, bool makeReferenceCurve
             if (mc != null)
             {
                 InternalSetCurveElement(mc);
-                if (!InternalSetSketchPlaneFromCurve(crv))
+                if (InternalSetSketchPlaneFromCurve(crv))
                 {
                     InternalSetCurve(crv);
-                    return;
```

Comment:
I still don't understand how the logic for case 2 here works, can you explain it to me?

the cases I see:

1. we identify that an old element existed, and we attempt to rebind, if we successfully set the curve and the plane, we do this and we return - I believe this will trigger a document modification event, and call the constructor again, we'll go into this method again, and determine we don't need to set the plane, - so we shouldn't modify the curve, and I think the loop will end here. Seems okay.

2. we identify that an old element existed, and attempt to rebind, we don't need to reset the plane - so we return false, and now... we never set the curve.... now we just return from the constructor and nothing has changed. Can you verify that this isn't an issue. I am thinking about the return statement on line 272, where `needsSketchPlaneReset` will still be false, and we won't have reset the plane, we won't set the curve in any case that I can see.

3. We identify an old element existed, and attempt to rebind, we can't find a reset method and need to reset the plane, so we return true set the curve on the model curve... will this work or throw an exception?

I guess I am just concerned because it doesn't seem there is a way for this code to hit the initial branch that creates a new model curve in the case we can't rebind. Can you please test this code out on some different curves in Dynamo, and move adjust the inputs to force rebinding if you have not already. (not just lines from grids)